sockeye/config.py

```diff
@@ -29,44 +29,14 @@ def __init__(cls, name, bases, kwds):
         super().__init__(name, bases, new_kwds)
 
 
+@dataclass
 class Config(yaml.YAMLObject, metaclass=TaggedYamlObjectMetaclass):
     """
     Base configuration object YAML (de-)serialization.
     Actual Configuration should subclass this object.
     """
     yaml_loader = yaml.UnsafeLoader  # type: ignore
 
-    def __setattr__(self, key, value):
-        if value == self:
-            raise AttributeError("Cannot set self as attribute")
-        object.__setattr__(self, key, value)
-
-    def __setstate__(self, state):
-        """Pickle protocol implementation."""
-        # We first take the serialized state:
-        self.__dict__.update(state)
-        # Then we take the constructors default values for missing arguments in order to stay backwards compatible
-        # This way we can add parameters to Config objects and still load old models.
-        init_signature = inspect.signature(self.__init__)
```

Comment:
I see from the test `test_config_missing_attributes_filled_with_default` that this is now already supported out of the box, neat!

Reply:
yes, pretty cool!